File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/lancamento/dto/LancamentoFinanceiroGridDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.lancamento.dto;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.atendimento.lancamento.entity.LancamentoFinanceiroSituacaoEnum;
import br.com.grupopipa.gestaointegrada.atendimento.lancamento.entity.LancamentoFinanceiroStatusFinanceiroEnum;
import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter
@Setter
public class LancamentoFinanceiroGridDTO implements GridDTO {

    private UUID id;
    private Long atendimentoNumero;
    private LocalDate dataAtendimento;
    private String pacienteNome;
    private String convenioNome;
    private BigDecimal valorTotal;
    private LancamentoFinanceiroSituacaoEnum situacao;
    private LancamentoFinanceiroStatusFinanceiroEnum statusFinanceiro;
    private int procedimentosCount;
    private LocalDateTime createdAt;
    private Boolean deleted;
}

File: src/frontend/src/app/components/atendimento/lancamento/model/lancamento-financeiro-grid-dto.ts
import { Exclude, Expose } from 'class-transformer';
import { LancamentoFinanceiroSituacao } from './lancamento-financeiro-situacao.enum';
import { LancamentoFinanceiroStatusFinanceiro } from './lancamento-financeiro-status-financeiro.enum';

@Exclude()
export class LancamentoFinanceiroGridDTO {
  @Expose() id?: string;
  @Expose() atendimentoNumero?: number;
  @Expose() dataAtendimento?: string;
  @Expose() pacienteNome?: string;
  @Expose() convenioNome?: string;
  @Expose() valorTotal?: number;
  @Expose() situacao?: LancamentoFinanceiroSituacao;
  @Expose() statusFinanceiro?: LancamentoFinanceiroStatusFinanceiro;
  @Expose() procedimentosCount?: number;
  @Expose() createdAt?: Date;
  @Expose() deleted?: boolean;
}
